util/bloom.cc

```diff
@@ -144,13 +144,28 @@ class FullFilterBitsReader : public FilterBitsReader {
       : data_(const_cast<char*>(contents.data())),
         data_len_(static_cast<uint32_t>(contents.size())),
         num_probes_(0),
-        num_lines_(0) {
+        num_lines_(0),
+        log2_cache_line_size_(0) {
     assert(data_);
     GetFilterMeta(contents, &num_probes_, &num_lines_);
     // Sanitize broken parameter
     if (num_lines_ != 0 && (data_len_-5) % num_lines_ != 0) {
       num_lines_ = 0;
       num_probes_ = 0;
+    } else if (num_lines_ != 0) {
+      while (true) {
+        uint32_t num_lines_at_curr_cache_size =
+            (data_len_ - 5) >> log2_cache_line_size_;
+        if (num_lines_at_curr_cache_size == 0) {
+          // The cache line size seems not a power of two. It's not supported
+          // and indicates a corruption that we can't report from here.
+          abort();
```

Comment:
thoughts on this? We could alternatively have an `assert(false)` so we can catch it in testing, but reset the parameters to zero so production builds would just not use the bloom filter. I went with this approach to make debugging easier if it's needed.

Reply:
If this could happen due to corruption of on-disk data, maybe its better to not abort() and instead let the production build not use the filter. Its still better than the old behavior of returning an incorrect result.

Reply:
Sure, let me try to add some logic so we can get Status::Corrupted back to the user.

Reply:
Never mind, propagating the corruption requires changing too much code. Let's disable the bloom filter.